src/ate/test_programs/dut_lib/dut_lib.cc

```diff
@@ -47,9 +47,10 @@ void DutLib::DutConsoleWaitForRx(const char* msg, uint64_t timeout_ms) {
 std::string DutLib::DutConsoleRx(bool quiet, uint64_t timeout_ms) {
   LOG(INFO) << "in DutLib::DutConsoleRx";
   size_t msg_size = kMaxRxMsgSizeInBytes;
-  OtLibConsoleRx(transport_, quiet, timeout_ms,
-                 reinterpret_cast<uint8_t*>(console_msg_buf_), &msg_size);
-  std::string result(console_msg_buf_, msg_size);
+  std::string result(kMaxRxMsgSizeInBytes, '\0');
+  OtLibConsoleRx(transport_, quiet, timeout_ms, (uint8_t*)result.data(),
```

Comment:
Can you perform this cast with const_cast: `const_cast<uint8_t*>(result.data())` ?

Does `&result[0]` work without a const-cast?

Reply:
`&result[0]` does not work unfortunately.

Also, the first `const_cast` does not work either.

What is needed is `reinterpret_cast<uint8_t*>(const_cast<char*>(result.data())),`. I added this instead.